src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -12807,119 +12807,127 @@ private async Task WriteXmlFeed(XmlDataFeed feed, TdsParserStateObject stateObj,
         private async Task WriteTextFeed(TextDataFeed feed, Encoding encoding, bool needBom, TdsParserStateObject stateObj, int size, bool useReadBlock)
         {
             Debug.Assert(encoding == null || !needBom);
-            char[] inBuff = new char[constTextBufferSize];
+            char[] inBuff = ArrayPool<char>.Shared.Rent(constTextBufferSize);
```

Comment:
For something like this, can we just do a stackalloc?

Reply:
That's unsafe territory... I don't think I'm the right person for that :)

Reply:
Understandable. I think we *could* consider changing this to stackalloc, but it would also require changing the type of the variable, and any usages of the variable, and we'd need to consider the size we're allocating... etc etc. So, yes, I see how it could be unsafe while not technically `unsafe` :)